mod/groups/start.php

```diff
@@ -357,6 +355,82 @@ function groups_activity_owner_block_menu($hook, $type, $return, $params) {
 	return $return;
 }
 
+/**
+ * Returns a menu item for leaving a group
+ *
+ * @param \ElggGroup $group Group to leave
+ * @param \ElggUser  $user  User to check leave action for
+ *
+ * @return \ElggMenuItem|false
+ */
+function groups_get_group_leave_menu_item(\ElggGroup $group, $user = null) {
+	if (!$group instanceof \ElggGroup) {
+		return false;
+	}
+	
+	if (is_null($user)) {
+		$user = elgg_get_logged_in_user_entity();
+	}
+	
+	if (empty($user)) {
+		return false;
+	}
+	
+	if (!$group->isMember($user) || ($group->owner_guid === $user->guid)) {
+		// a member can leave a group if he/she doesn't own it
+		return false;
+	}
+	
+	return \ElggMenuItem::factory([
+		'name' => 'groups:leave',
+		'icon' => 'sign-out',
+		'text' => elgg_echo('groups:leave'),
+		'href' => "action/groups/leave?group_guid={$group->guid}",
```

Comment:
add user_guid to the url as apparently this function is allowed for another user than current user

Reply:
elgg_http_add_url_query_elements